src/gui/mzroll/pollyelmaveninterface.cpp

```diff
@@ -966,6 +1117,107 @@ QStringList PollyElmavenInterfaceDialog::_prepareFilesToUpload(QDir qdir,
     return filenames;
 }
 
+QStringList
+PollyElmavenInterfaceDialog::_prepareSessionFiles(QString datetimestamp)
+{
+    QStringList filenames;
+
+    // write a JSON and CSV file for each peak table
+    QList<QPointer<TableDockWidget>> peakTableList =
+        _mainwindow->getPeakTableList();
+    peakTableList.append(_mainwindow->getBookmarkedPeaks());
+    for (TableDockWidget* peakTable : peakTableList) {
+        if (peakTable->getGroups().isEmpty())
+            continue;
+
+        QString tableName = TableDockWidget::getTitleForId(peakTable->tableId);
+        tableName.replace(" ", "_"); // replace spaces with underscores
+        QString jsonFilename = _writeableTempDir
+                               + QDir::separator()
+                               + datetimestamp
+                               + "_"
+                               + tableName
+                               + ".json";
+        peakTable->exportJsonToPolly(_writeableTempDir, jsonFilename, false);
+        filenames.append(jsonFilename);
+
+        QCoreApplication::processEvents();
+
+        QString csvFilename = datetimestamp + "_" + tableName + ".csv";
+        peakTable->treeWidget->selectAll();
+        peakTable->prepareDataForPolly(_writeableTempDir,
```

Comment:
Group Summary is helpful for LCMS and DDA data but Polly requires Peak detailed format for MRM datasets. Maybe send both formats?

Reply:
I will just send both, in case the other is also needed for something specific.